### PR TITLE
ci: fix pypi release

### PR DESCRIPTION
**Pull request by universalmind303**

### CODE REVIEW

File: .github/workflows/python-release.yaml [hunk 1]
```diff
@@ -126,7 +126,7 @@ jobs:
       - name: Build wheels
         uses: PyO3/maturin-action@v1
         with:
-          target: ${{ matrix.target }}
+          target: ${{ matrix.target }}-apple-darwin
```
Comment (universalmind303) [inline .github/workflows/python-release.yaml:129]: AFAIK, `x86_64` and `aarch64` arent valid targets, they should be they fully qualified ones `aarch64-apple-darwin`

https://www.maturin.rs/config#target-specific-maturin-options

Comment (greyscaled) [reply .github/workflows/python-release.yaml:129]: Yea, I noticed this in #2556 as well.

File: .github/workflows/python-release.yaml [hunk 2]
```diff
@@ -108,7 +108,7 @@ jobs:
 
 
   macos:
-    runs-on: macos-12-xl
+    runs-on: macos-14-xlarge
```
Comment (greyscaled) [inline .github/workflows/python-release.yaml:111]: cc @tychoish & @scsmithr you have context from #2543

**Review by greyscaled — APPROVED** [review]

File: .github/workflows/python-release.yaml (hunk 2 above)
Comment (tychoish) [reply .github/workflows/python-release.yaml]: we should definitely just mirror whatever we're doing for the CLI.

my logic for choosing the 13 was just "use the oldest and smallest thing that works.

File: .github/workflows/python-release.yaml (hunk 1 above)
Comment (tychoish) [reply .github/workflows/python-release.yaml:129]: is this a recent change?

**Review by tychoish — APPROVED** [review]

File: .github/workflows/python-release.yaml [hunk 3]
```diff
@@ -113,13 +111,17 @@ jobs:
...
       matrix:
-        target: [x86_64, aarch64]
+        config:
+        - target: x86_64-apple-darwin
+          runner: macos-13-large
+        - target: aarch64-apple-darwin
+          runner: macos-13-xlarge
 
-    name: Build Python Bindings (macos ${{ matrix.target }})
+    name: Build Python Bindings (macos ${{ matrix.config.target }})
```
Comment (greyscaled) [inline .github/workflows/python-release.yaml:124]: I think we need to make this change again below (line 154 if I'm reading the diff right) where we post to slack

**Pull request merged**